src/storage/src/hummock/iterator/merge_inner.rs

```diff
@@ -88,22 +90,122 @@ impl<'a, const DIRECTION: usize> MergeIteratorInner<'a, DIRECTION> {
 
         self.heap = self
             .unused_iters
-            .drain_filter(|i| i.is_valid())
-            .map(Node)
+            .drain_filter(|i| i.iter.is_valid())
             .collect();
     }
 }
 
 #[async_trait]
```

Comment:
Note that the `async_trait` itself may also lead to dynamic dispatching. We may investigate this later.